lib/data_service/common.dart

```diff
@@ -26,12 +56,45 @@ abstract class TraktService {
     var jsonData = JSON.decode(responseText);
     var movies = new Set<Movie>();
     for (var watchlistData in jsonData) {
-      var movieData = watchlistData["movie"];
-      var slug = movieData["slug"];
-      var url = "https://trakt.tv/movies/${slug}";
-      movies.add(new Movie(movieData["title"], movieData["year"], url));
+      movies.add(_makeMovie(watchlistData["movie"]));
     }
 
     return movies;
   }
+
+  Future<Map<int, Set<Movie>>> ratings(String username) async {
+    var responseText = await fetchRatings(username);
+    // TODO: check if response succeeded.
+    var jsonData = JSON.decode(responseText);
+    var ratings = new Map<int, Set<Movie>>();
+    for (var ratingData in jsonData) {
+      var rating = ratingData["rating"];
+      var movie = _makeMovie(ratingData["movie"]);
+      ratings.putIfAbsent(rating, () => new Set()).add(movie);
+    }
+
+    return ratings;
+  }
+
+  Future<Map<Movie, DateTime>> lastWatched(String username) async {
+    var responseText = await fetchLastWatched(username);
+    // TODO: check if response succeeded.
+    var jsonData = JSON.decode(responseText);
+    var lastWatchedMap = new Map<Movie, DateTime>();
+    for (var lastWatchedData in jsonData) {
+      var lastWatched = DateTime.parse(lastWatchedData["last_watched_at"]);
+      var movie = _makeMovie(lastWatchedData["movie"]);
+      lastWatchedMap[movie] = lastWatched;
+    }
+
+    return lastWatchedMap;
+  }
+
+  Future<UserData> fetchData(String username) async {
```

Comment:
Add a comment that this is not the most concurrent solution; would need to map all individual data call and then reduce back down to `UserData`.